### PR TITLE
Fix/asb 1963 submission filename

### PR DESCRIPTION
**Pull request by JamesP-KC**
## Describe your changes

ASB-1968, ASB-1818

Updates to the `POST /api/v1/submissions/{submissionId}` and `POST /api/v1/submissions/{submissionId}/complete` endpoints

- aadb0c2d9bc8e012608937bdacf25f90b97fa4b2 simplify the file naming conventions used when uploading files to blob storage container. The PDF file generated for a written rep will have the name `Written-Representation.pdf` (previously had the submissionID included in the file name). A user uploaded file will be uploaded as-is and the file name will not be changed. 
- c53c62916d601274f98ff7c247ca4260d4c6630f fix issue with api not returning a 400 error if the `caseReference` property was missing from the request body but the `email` _was_ included. Updated the error message to be: `"must include both 'email' and 'caseReference'"`.

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/applications-service-api/src/controllers/submissions.js [hunk 1]
```diff
@@ -10,11 +11,21 @@ const createSubmissionController = async (req, res) => {
...
 const completeSubmissionController = async (req, res) => {
-	await completeSubmission({
+	const submissionDetails = {
 		submissionId: req.params.submissionId,
 		caseReference: req.body?.caseReference,
 		email: req.body?.email
-	});
+	};
+
+	if (
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/src/controllers/submissions.js:20]: If there is no body or empty body then I think the existing Front end call for NI integration would get past this check and work.  Is that by design?

I think we may need a front end change to send the body but perhaps that is not for this ticket.

Comment (JamesP-KC) [reply packages/applications-service-api/src/controllers/submissions.js:20]: Yeah, that's intentional. The existing NI call has no body so I've made email and caseref optional (but if you provide one, then you must provide the other)

**Review by Philip-Philippou1 — APPROVED** [review]

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**